sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala

```diff
@@ -159,36 +159,37 @@ class HadoopTableReader(
     def verifyPartitionPath(
         partitionToDeserializer: Map[HivePartition, Class[_ <: Deserializer]]):
         Map[HivePartition, Class[_ <: Deserializer]] = {
-      if (!sparkSession.sessionState.conf.verifyPartitionPath) {
-        partitionToDeserializer
-      } else {
-        var existPathSet = collection.mutable.Set[String]()
-        var pathPatternSet = collection.mutable.Set[String]()
-        partitionToDeserializer.filter {
-          case (partition, partDeserializer) =>
-            def updateExistPathSetByPathPattern(pathPatternStr: String) {
-              val pathPattern = new Path(pathPatternStr)
-              val fs = pathPattern.getFileSystem(hadoopConf)
-              val matches = fs.globStatus(pathPattern)
-              matches.foreach(fileStatus => existPathSet += fileStatus.getPath.toString)
-            }
-            // convert  /demo/data/year/month/day  to  /demo/data/*/*/*/
-            def getPathPatternByPath(parNum: Int, tempPath: Path): String = {
+      var existPathSet = collection.mutable.Set[String]()
+      var pathPatternSet = collection.mutable.Set[String]()
+      partitionToDeserializer.filter {
+        case (partition, partDeserializer) =>
+          def updateExistPathSetByPathPattern(pathPatternStr: String) {
+            val pathPattern = new Path(pathPatternStr)
+            val fs = pathPattern.getFileSystem(hadoopConf)
+            val matches = fs.globStatus(pathPattern)
+            matches.foreach(fileStatus => existPathSet += fileStatus.getPath.toString)
+          }
+          // convert  /demo/data/year/month/day  to  /demo/data/*/*/*/
+          def getPathPatternByPath(parNum: Int, tempPath: Path, partitionName: String): String = {
+            // if the partition path does not end with partition name, we should not
```

Comment:
if the partition location has been altered to another location, we should not do this pattern, or we will list pattern files which does not belong to the partition